Common++/header/IpAddress.h

```diff
@@ -687,6 +687,322 @@ namespace pcpp
 	};
+
+
+	/**
+	 * @class IPNetwork
+	 * A class representing version independent IP network definition, both IPv4 and IPv6 are included
+	 */
+	class IPNetwork
+	{
+	public:
+		/**
+		 * A constructor that creates an instance of the class out of an address representing the network prefix
+		 * and a prefix length
+		 * @param address An address representing the network prefix. If the address is invalid std::invalid_argument
+		 * exception is thrown
+		 * @param prefixLen A number representing the prefix length. If the value isn't in the range allowed for the
+		 * network (0 - 32 for IPv4 networks or 0 - 128 for IPv6 networks) and std::invalid_argument exception is thrown
+		 */
+		IPNetwork(const IPAddress& address, uint8_t prefixLen)
+		{
+			if (address.isIPv4())
+			{
+				m_IPv4Network = new IPv4Network(address.getIPv4(), prefixLen);
+				m_IPv6Network = nullptr;
+			}
+			else
+			{
+				m_IPv6Network = new IPv6Network(address.getIPv6(), prefixLen);
+				m_IPv4Network = nullptr;
+			}
+		}
+
+		/**
+		 * A constructor that creates an instance of the class out of an address representing the network prefix
+		 * and a netmask
+		 * @param address An address representing the network prefix. If the address is invalid std::invalid_argument
+		 * exception is thrown
+		 * @param netmask A string representing a netmask in valid format, for example: ffff:ffff:: for IPv6 networks
+		 * or 255.255.0.0 for IPv4 networks.
+		 * Please notice that netmasks that start with zeros are invalid, for example: 0:ffff:: or 0.255.255.255.
+		 * The only netmask starting with zeros that is valid is all zeros (:: or 0.0.0.0).
+		 * If the netmask is invalid std::invalid_argument exception is thrown
+		 */
+		IPNetwork(const IPAddress& address, const std::string& netmask)
+		{
+			if (address.isIPv4())
+			{
+				m_IPv4Network = new IPv4Network(address.getIPv4(), netmask);
+				m_IPv6Network = nullptr;
+			}
+			else
+			{
+				m_IPv6Network = new IPv6Network(address.getIPv6(), netmask);
+				m_IPv4Network = nullptr;
+			}
+		}
+
+		/**
+		 * A constructor that creates an instance of the class out of a string representing the network prefix and
+		 * a prefix length or a netmask
+		 * @param addressAndNetmask A string in one of these formats:
+		 *  - IP_ADDRESS/Y where IP_ADDRESS is a valid IP address representing the network prefix and Y is
+		 *    a number representing the network prefix
+		 *  - IP_ADDRESS/NETMASK where IP_ADDRESS is a valid IP address representing the network prefix and NETMASK
+		 *    is a valid netmask for this type of network (IPv4 or IPv6 network)
+		 *  For any invalid value std::invalid_argument is thrown
+		 */
+		IPNetwork(const std::string& addressAndNetmask)
+		{
+			try
+			{
+				m_IPv4Network = new IPv4Network(addressAndNetmask);
+				m_IPv6Network = nullptr;
+			}
+			catch (const std::invalid_argument&)
+			{
+				m_IPv6Network = new IPv6Network(addressAndNetmask);
```

Comment:
Can the IPv6Network also raise a signal ?

Reply:
yes, it can raise an exception, but it means the string is not a valid IPv4 nor an IPv6 network, hence the exception should be thrown outside to the user